lib/rpush/daemon/pushy/delivery.rb

```diff
@@ -0,0 +1,90 @@
+module Rpush
+  module Daemon
+    module Pushy
+      class Delivery < Rpush::Daemon::Delivery
+        include MultiJsonHelper
+
+        attr_reader :http, :notification, :batch, :pushy_uri
+
+        def initialize(app, http, notification, batch)
+          @http = http
+          @notification = notification
+          @batch = batch
+          @pushy_uri = URI.parse("https://api.pushy.me/push?api_key=#{app.api_key}")
+        end
+
+        def perform
+          response = send_request
+          process_response(response)
+        rescue SocketError => error
+          mark_retryable(notification, Time.now + 10.seconds, error)
```

Comment:
Will this create possibility of an endless retry loop every 10 seconds on socket failure? Are there any max count limit somewhere that will terminate the retries and raise error in logs?

Reply:
There is no any max count limit. All implementations follows this approach, for example, look at https://github.com/rpush/rpush/blob/master/lib/rpush/daemon/gcm/delivery.rb#L23

Reply:
ok